lib/opentelemetry/wrapper.ts

```diff
@@ -41,10 +53,99 @@ export async function withMeasurement<T>(
 		} catch (instrumentationError) {
 			captureError(logger, instrumentationError, "instrumentation failed");
 		}
-
 		return result;
 	} catch (error) {
-		captureError(logger, error, `[${externalServiceName}] operation failed`);
+		captureError(logger, error, "operation failed");
 		throw error;
 	}
 }
+
+const APICallBasedService = {
+	Unstructured: ExternalServiceName.Unstructured,
+	Tavily: ExternalServiceName.Tavily,
+	Firecrawl: ExternalServiceName.Firecrawl,
+} as const;
+
+const TokenBasedService = {
+	OpenAI: ExternalServiceName.OpenAI,
+} as const;
+
+export function withCountMeasurement<T>(
+	logger: OtelLoggerWrapper,
+	operation: () => Promise<T>,
+	externalServiceName: typeof APICallBasedService.Unstructured,
+	measurementStartTime: number | undefined,
+	strategy: Strategy,
+): Promise<T>;
+export function withCountMeasurement<T>(
+	logger: OtelLoggerWrapper,
+	operation: () => Promise<T>,
+	externalServiceName:
+		| typeof APICallBasedService.Tavily
+		| typeof APICallBasedService.Firecrawl,
+	measurementStartTime?: number,
+): Promise<T>;
+export function withCountMeasurement<T>(
+	logger: OtelLoggerWrapper,
+	operation: () => Promise<T>,
+	externalServiceName: (typeof APICallBasedService)[keyof typeof APICallBasedService],
+	measurementStartTime?: number,
+	strategy?: Strategy,
+): Promise<T> {
+	const measurement: MeasurementSchema<T> = (
+		_result,
+		duration,
+		measurementScope,
+		isR06User,
+	): RequestCountSchema => {
+		const baseMetrics = {
+			duration,
+			measurementScope,
+			isR06User,
+			requestCount: 1,
+		};
+
+		if (externalServiceName === APICallBasedService.Unstructured) {
+			if (!strategy) {
+				logger.error(
+					new Error("'strategy' is required for Unstructured service"),
```

Comment:
If required parameter `strategy` is missing, emit `ERROR` log but do not throw error to avoid interfere business logic